prestodb/client.py

```diff
@@ -329,11 +334,23 @@ def get_url(self, path):
     @property
     def statement_url(self):
         # type: () -> Text
+        if self._prefix:
+            return self.get_url(f"{self._prefix}{constants.URL_STATEMENT_PATH}")
+
         return self.get_url(constants.URL_STATEMENT_PATH)
 
     @property
     def next_uri(self):
         # type: () -> Text
+        if self._prefix:
+            logger.debug(f"Updating url {self._next_uri} with prefix {self._prefix}")
+            pattern = r"(http[s]?://[^/]+)(/.*)"
+            try:
+                return re.sub(pattern, r"\1" + self._prefix + r"\2", self._next_uri)
+            except TypeError:
+                logger.error("Unable to update url with prefix. "
+                             "Continuing without prefix.")
```

Comment:
Instead of failing during query execution, does it make sense to raise the error during the initialization if the `_prefix` provided doesn't match the regex?